addon/services/activity-watcher.ts

```diff
@@ -0,0 +1,152 @@
+import Service from '@ember/service';
+import { inject as service } from '@ember/service';
+
+import { Observable } from '@upfluence/hyperevents/helpers/observable';
+import EventsService, { ResourceEvent, prefixPath } from '@upfluence/hyperevents/services/events-service';
+import ToastService, { ToastOptions } from '@upfluence/oss-components/services/toast';
+
+type NotificationEvent = {
+  resource: string;
+  payload: {
+    notification_type: string;
+    timestamp: number;
+    data: any;
+  };
+};
+
+type RenderedNotification = {
+  title: string;
+  message: string;
+  type: string;
+};
+
+function renderNotificationMessage(message: string): RenderedNotification {
+  return {
+    title: `<i class="toast-title__icon upf-icon upf-icon--messages"></i>`,
+    message: message,
+    type: 'info'
+  };
+}
+
+function renderNotificationErrorMessage(message: string): RenderedNotification {
+  return {
+    title: `<i class="fa fa-info-circle" aria-hidden="true"></i>`,
+    message: message,
+    type: 'error'
+  };
+}
+
+function renderNotificationMessageWithAvatar(avatarUrl: string, message: string): RenderedNotification {
+  return {
+    title: `<img class="toast-title__avatar" src="${avatarUrl}" />
+     <i class="toast-title__icon upf-icon upf-icon--messages"></i>`,
+    message: message,
+    type: 'info'
+  };
+}
+
+const toastOpts: ToastOptions = { timeout: 0 };
+
+export default class ActivityWatcher extends Service.extend({}) {
+  @service declare eventsService: EventsService;
+  @service declare toast: ToastService;
+
+  private declare _observer: Observable<ResourceEvent> | null;
+
+  public watch() {
```

Comment:
you don't need to mark the methods as `public` :p 
also, can you add `: void` return type